### PR TITLE
Expose ZipArchiveEntry.Crc32 and fix crc bug

### PR DESCRIPTION
**Pull request by ianhays**
- Exposes ZipArchiveEntry.Crc32 field
- Fixes a bug in our CRC calculation
- Adds tests for the crc32 field

Progression of https://github.com/dotnet/corefx/pull/25204 by @reaction1989

resolves https://github.com/dotnet/corefx/issues/22292, https://github.com/dotnet/corefx/issues/25803

cc: @stephentoub @ViktorHofer @livarcocc

### CODE REVIEW

**Comment by ianhays** [comment]
@dotnet-bot test NETFX x86 Release Build please

File: src/Common/tests/System/IO/Compression/ZipTestHelper.cs [hunk 1]
```diff
@@ -336,6 +340,14 @@ private static void ItemEqual(string[] actualList, List<FileData> expectedList,
                                         entryStream.Write(new ReadOnlySpan<byte>(buffer, 0, bytesRead));
                                     }
                                 }
+                                else if (writeInChunks)
```
Comment (danmosemsft) [inline src/Common/tests/System/IO/Compression/ZipTestHelper.cs]: can you explain this writeInChunks option that only tests are using..

Comment (danmosemsft) [reply src/Common/tests/System/IO/Compression/ZipTestHelper.cs]: if it's to simulate an innstallStream that only returns <= 5 chars each read, which exercises the CRC calculation, could you pass in such a stream in the test instead?

Comment (ianhays) [reply src/Common/tests/System/IO/Compression/ZipTestHelper.cs]: Sure! This was added to hit the failure case in https://github.com/dotnet/corefx/issues/25803 where we were only testing one version of write: the one without an offset. So we didn't have any test coverage where Write was called with a nonzero offset which led to us missing a bug.

Note that this is in the `ZipTestHelper` file.

Comment (danmosemsft) [reply src/Common/tests/System/IO/Compression/ZipTestHelper.cs]: oh disregard, I see this is a test method. Still could use a comment.

Comment (ianhays) [reply src/Common/tests/System/IO/Compression/ZipTestHelper.cs]: > Still could use a comment.

Sure thing :)

Comment (danmosemsft) [reply src/Common/tests/System/IO/Compression/ZipTestHelper.cs]: I see it exposes the bug you fixed.

File: src/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs [hunk 2]
```diff
@@ -145,6 +145,9 @@ internal ZipArchiveEntry(ZipArchive archive, string entryName)
         /// </summary>
         public ZipArchive Archive => _archive;
 
+        [CLSCompliant(false)]
```
Comment (danmosemsft) [inline src/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs]: There are other uint properties in this library that don't have CLSCompliant. Do we need to bother with it.

Comment (ianhays) [reply src/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs]: I believe so, based on this comment in the original PR: https://github.com/dotnet/corefx/pull/25204#discussion_r150416742

**Comment by natemcmaster** [comment]
Any estimate on getting this in soon? Our build is blocked on this CRC issue. Its affecting our build output, making \*.nupkg files unusable on some clients.

**Pull request merged**